tests/queries/0_stateless/03373_write_to_merge_table.sql

```diff
@@ -0,0 +1,39 @@
+DROP TABLE IF EXISTS test03373_db.test03373_table_1;
+DROP TABLE IF EXISTS test03373_db.test03373_table_2;
+DROP TABLE IF EXISTS test03373_db.test03373_merge_ro;
+DROP TABLE IF EXISTS test03373_db.test03373_merge_wr_1;
+DROP TABLE IF EXISTS test03373_db.test03373_merge_wr_2;
+DROP TABLE IF EXISTS test03373_db.test03373_merge_wr_3;
+DROP DATABASE IF EXISTS test03373_db;
+
+CREATE DATABASE test03373_db;
+
+CREATE TABLE test03373_db.test03373_table_1 (key UInt32, value UInt32) ENGINE=MergeTree() ORDER BY key;
+CREATE TABLE test03373_db.test03373_table_2 (key UInt32, value UInt32) ENGINE=MergeTree() ORDER BY key;
+
+CREATE TABLE test03373_db.test03373_merge_ro (key UInt32, value UInt32) ENGINE=Merge(test03373_db, 'test03373_table_\d+');
+
+CREATE TABLE test03373_db.test03373_merge_wr_1 (key UInt32, value UInt32) ENGINE=Merge(test03373_db, 'test03373_table_\d+', test03373_table_2);
+CREATE TABLE test03373_db.test03373_merge_wr_2 (key UInt32, value UInt32) ENGINE=Merge(test03373_db, 'test03373_table_\d+', test03373_db.test03373_table_2);
+CREATE TABLE test03373_db.test03373_merge_wr_3 (key UInt32, value UInt32) ENGINE=Merge(REGEXP('test03373_.*'), 'test03373_table_\d+', test03373_db.test03373_table_2);
+
```

Comment:
Doesn't it make sense to add tests
1. where an underlying table we plan to write to does not exist at the moment of creating a table with Engine Merge, in other words
CREATE TABLE test03373_db.test03373_merge_wr_42 (key UInt32, value UInt32) ENGINE=Merge(test03373_db, 'test03373_table_\d+', not_created yet);
CREATE TABLE not_created yet ...
2. where no grant to write into underlying table
3. where no grant to write into Merge table
?

Reply:
2. Write is success.
3. Error ' Not enough privileges.'

I'm not sure what is right in case 2 - grant to insert into merge table is enough or not.
Need some suggest.

Reply:
@ianton-ru , based on my experience, we have to obey grants, so grant for underlying table is mandatory. As well as row level security BTW.

But we have to double check it e.g. with @filimonov

Reply:
> By the way - right now grants on select work with the same way - when user has grant to read from merge table but does not have grant to read from source table, he can get data from merge table.

Interesting.

Not sure if it was the same at the moment of developing
https://github.com/ClickHouse/ClickHouse/pull/50209

Reply:
Added check write access